### PR TITLE
Add log when user adds or removes a private link [OSF-7309]

### PR DESCRIPTION
**Pull request by erinspace**
## Purpose
Add logs whenever a view-only link is created on a project or component. Currently, a user could create a view-only link, share that with others, allowing them to access the project. The user could then remove the VOL and his collaborators would never know that the project's contents had been shared. 

Create new logs that say when a normal or an anonymous view only link is added or removed.

screen shot 2017-02-27 at 12 06 32 pm

## Changes

- Add `VIEW_ONLY_LINK_ADDED` and `VIEW_ONLY_LINK_REMOVED` to the `NodeLog` model
- Add `anonymous_link` to the NodeLog serializer
- Add log when node link is added or removed
- Tests to make sure logs are added
- Log text added to log actions list
- Log kwarg `anonymous_link` parsed in `LogTextParser.js`

## Side effects

None anticipated

## Ticket
https://openscience.atlassian.net/browse/OSF-7309

### CODE REVIEW

**Comment by brianjgeiger** [comment]
Fix merge conflicts when you have a moment, please.

File: website/project/views/node.py [hunk 1]
```diff
@@ -605,6 +606,21 @@ def remove_private_link(*args, **kwargs):
         link = PrivateLink.load(link_id)
         link.is_deleted = True
         link.save()
+
+        for node in link.nodes.all():
```
Comment (erinspace) [inline website/project/views/node.py]: As for the django-y ness, normally I'd be worried about this here, but I do need to add a log to each node, so can't use something like `values` or `values_list` I don't think....

File: website/project/views/node.py [hunk 2]
```diff
@@ -596,6 +597,21 @@ def remove_private_link(*args, **kwargs):
...
+            log_dict = {
+                'project': node.parent_id,
```
Comment (sloria) [inline website/project/views/node.py]: Isn't `parent_id` the pk of the parent, not the GUID?

Comment (erinspace) [inline website/project/views/node.py]: [Seems to be the _id?](https://github.com/CenterForOpenScience/osf.io/blob/develop/osf/models/node.py#L967). Also, I was following the example of [other node logs that use node.parent_id](https://github.com/CenterForOpenScience/osf.io/blob/develop/osf/models/node.py#L1081)

Comment (sloria) [inline website/project/views/node.py]: Ah, cool cool. Just making sure.

**Comment by erinspace** [comment]
@brianjgeiger merge conflict free!

**Pull request merged**